src/Tizen.System.PowerUsage/PowerUsage/PowerUsage.cs

```diff
@@ -0,0 +1,242 @@
+                            Log.Error(PowerUsageErrorFactory.LogTag, $"Error getting battery usage for {type}");
+                        }
+                    }
+                }
+                else
+                {
+                    Log.Error(PowerUsageErrorFactory.LogTag, "Power usage resource types parameter is empty");
+                    throw new ArgumentNullException(nameof(rtypes));
+                }
+            }
+            finally
+            {
+                PowerUsageError ret = (PowerUsageError)Interop.PowerUsage.BatteryUsageDataDestroy(dataHandle);
+                if (ret != PowerUsageError.None)
+                {
+                    Log.Error(PowerUsageErrorFactory.LogTag, "Error in Destroy handle, " + ret);
+                }
+            }
+            return batteryUsage;
+        }
+
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the specific resource for the given application in custom interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="rtype">Identifier of resource type. BLE, WiFi, CPU etc.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the specific resource for the given application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        /// <exception cref="ArgumentException">When an invalid parameter value is set.</exception>
+        /// <exception cref="UnauthorizedAccessException">If the privilege is not set.</exception>
+        /// <exception cref="InvalidOperationException">In case of any system error.</exception>
+        /// <exception cref="NotSupportedException">In case power usage is not supported</exception>
+        public static double GetPowerUsage(string appID, PowerUsageResourceType rtype, DateTime start, DateTime end)
+        {
+            double batteryUsage = 0;
+            DateTimeOffset startTime = DateTime.SpecifyKind(start, DateTimeKind.Utc);
+            DateTimeOffset endTime = DateTime.SpecifyKind(end, DateTimeKind.Utc);
+            PowerUsageError ret = (PowerUsageError)Interop.PowerUsage.GetPowerUsageByAppPerResource(appID, rtype, startTime.ToUnixTimeSeconds(), endTime.ToUnixTimeSeconds(), out batteryUsage);
+            if (ret != PowerUsageError.None)
+            {
+                if (ret == PowerUsageError.RecordNotFound)
+                {
+                    Log.Error(PowerUsageErrorFactory.LogTag, "Error PowerUsageResourceType is not supported");
+                    throw new ArgumentException($"{rtype} is not supproted", nameof(rtype));
```

Comment:
NotSupportedException or ArgumentException?

Reply:
NotSupportedException is used to check whether power usage is supported by the device or not.

In this case, if PowerUsageResourceType is not supported, then InvalidOperationException is thrown, but while using this api, user knows the resource type he is sending, so incase of unsupported resource, we are taking as wrong arguments and throwing Argument Exception